### DISCUSSION TITLE
Same widgets in frontend and backend ?

### DISCUSSION THREAD

**Discussion by Incremental92** [wintercms/winter#85] [Ideas]
Hello, I started a project 1,5 year ago and it's still not finished because what I designed with widgets, with Builder in the backend was not all available in the frontend (ie repeater).
Then I moved for the frontend to vue.js and started to use October as a headless CMS (missing GraphQL).
I would greatly love in Winter to have the same widgets (functionalities) both in backend and frontend, then not needing a Javascript frontend...
Long life to Winter !

**Comment by bennothommo** [comment]
@Incremental92 I've moved this into the Ideas discussion board.

Would you be able to clarify your request though - are you requesting that the functionality and widgets of the backend should be made available in the frontend (CMS) side as well?

**Reply by Incremental92** [reply]
Yes exactly.
As far as I understood, in October, it was different widgets and not all backend ones were adapted to frontend.

**Reply by bennothommo** [reply]
I'm not sure if we'll ever do this in core, simply because one of the advertised features of Winter CMS is that we don't impose any design or framework limitations on the CMS side - our widgets in the backend would force jQuery and Bootstrap on the front-end to work exactly as it does in the Backend.

As I understand it, there were a couple of plugins that provided this functionality for October, so you could look into that.

**Reply by Incremental92** [reply]
Well, I understand. But the plugins are paid plugins.
It confirm my choice :
- October for regular websites
- October as headless with Vue.js for specific business applications.
  October is fine with Builder to manage backend application and JWT Auth. What I miss is REST or GraphQL.

Do you confirm this choice ? I think Winter would really be better than Strapi kind of CMS for all use case.
There should be efficient plugins for Builder, JWT Auth, REST and GraphQL...
PS : as the learning curve for all these tools could be hard, I'd prefer a perfect unique "swiss knife"

**Reply by bennothommo** [reply]
@Incremental92 what you say is already achievable. You can set up Winter CMS with just the Backend and System modules (disabling the CMS module, effectively), to act as a headless administration panel. Then, you can create a plugin which provides a REST and/or GraphQL API for your front-end application. That way, you can build the front-end application in whichever framework and workflow you like.

**Reply by Incremental92** [reply]
@bennothommo the idea would be to avoid reinventing hot water.
I would love such plugins available in Winter !
Do you have plans for that ?
